File: web-service/src/main/java/Post/AssignmentSubmit.java
package Post;

import Files.UpFile;
import User.User;
import core.SEntity;

/**
 * a solution for an assignment submitted by a student
 */
public class AssignmentSubmit extends SEntity {

	protected int studentID, assignmetID;
	protected int grade;
	protected String studentNotes;
	protected UpFile[] files;

	public User fetchStudent() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public Assignment fetchAssignmet() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

}

File: web-service/src/models/post/assignment-submit.ts

/**
 * a solution for an assignment submitted by a student
 */
export class AssignmentSubmit extends SEntity {
	studentID: number;

	assignmetID: number;

	grade: number;

	studentNotes: string;

	files: UpFile[];

	public fetchStudent(): User {
		return null;
	}

	public fetchAssignmet(): Assignment {
		return null;
	}

	constructor() {
		super();
		this.studentID = 0;
		this.assignmetID = 0;
		this.grade = 0;
	}
}